seal.go

```diff
@@ -378,48 +379,77 @@ func SealKeyToTPM(tpm *TPMConnection, key []byte, keyPath string, params *KeyCre
 	return SealKeyToTPMMultiple(tpm, []*SealKeyRequest{{Key: key, Path: keyPath}}, params)
 }
 
-func updateKeyPCRProtectionPolicyCommon(tpm *tpm2.TPMContext, keyPath string, authData interface{}, pcrProfile *PCRProtectionProfile, session tpm2.SessionContext) error {
-	// Open the key data file
-	keyFile, err := os.Open(keyPath)
+func updateKeyPCRProtectionPolicyCommon(tpm *tpm2.TPMContext, keyPaths []string, authData interface{}, pcrProfile *PCRProtectionProfile, session tpm2.SessionContext) error {
+	var datas []*keyData
+	// Open the primary data file
+	keyFile, err := os.Open(keyPaths[0])
 	if err != nil {
 		return xerrors.Errorf("cannot open key data file: %w", err)
 	}
 	defer keyFile.Close()
 
-	data, authKey, pcrPolicyCounterPub, err := decodeAndValidateKeyData(tpm, keyFile, authData, session)
+	// Validate the primary file
+	primaryData, authKey, pcrPolicyCounterPub, err := decodeAndValidateKeyData(tpm, keyFile, authData, session)
 	if err != nil {
 		if isKeyFileError(err) {
 			return InvalidKeyFileError{err.Error()}
 		}
 		// FIXME: Turn the missing lock NV index in to ErrProvisioning
 		return xerrors.Errorf("cannot read and validate key data file: %w", err)
 	}
+	datas = append(datas, primaryData)
 
-	authPublicKey := data.staticPolicyData.authPublicKey
-	v0PinIndexAuthPolicies := data.staticPolicyData.v0PinIndexAuthPolicies
+	// Open and validate secondary files and make sure they are related
+	for _, p := range keyPaths[1:] {
+		keyFile, err := os.Open(p)
+		if err != nil {
+			return xerrors.Errorf("cannot open related key data file: %w", err)
+		}
+		defer keyFile.Close()
+
+		data, _, _, err := decodeAndValidateKeyData(tpm, keyFile, nil, session)
+		if err != nil {
+			if isKeyFileError(err) {
+				return InvalidKeyFileError{err.Error() + " (" + p + ")"}
+			}
+			// FIXME: Turn the missing lock NV index in to ErrProvisioning
+			return xerrors.Errorf("cannot read and validate related key data file: %w", err)
+		}
+		// The data is valid and the sealed key shares the same dynamic authorization policy
+		// signing key. Also verify that it has the same static authorization policy.
```

Comment:
This comment is not super clear to me, at least the order it mentions things is a bit odd. We don't have an explicit check about the signing key, OTOH below we do check the "static authorization policy." which implies the former? I would have expected the emphasis of the comment to be reversed, or I'm missing something

Reply:
The comment is wrong - I wrote it and then changed the code without updating the comment. I've fixed it now.